blockpage.lua

```diff
@@ -0,0 +1,153 @@
+--
+-- SPDX-FileCopyrightText: (c) 2024 Ring Zero Desenvolvimento de Software LTDA
+-- SPDX-License-Identifier: GPL-2.0-only
+--
+
+local socket = require("socket")
+local unpack = require("dome/unpack")
+local config = require("dome/config")
+
+local PACKET = socket.af.PACKET
+local RAW    = socket.sock.RAW
+
+local ETH_HLEN  = 14
+local ETH_P_ALL = 0x0003
+
+local IPPROTO_TCP = 0x06
+
+local RST = 1 << 2
+local ACK = 1 << 4
+
+local packer   = unpack.packer
+local unpacker = unpack.unpacker
+
+blockpage = {}
+
+-- https://datatracker.ietf.org/doc/html/rfc1141
+local function checksum(data, size)
+	local sum = 0
+	for i = 1, size, 2 do
+		sum = sum + string.unpack(">I2", data, i)
```

Comment:
any advantage on mixing `string.unpack` and `unpacker` instead only one or another?

Reply:
I just didn't think it was necessary to use `luadata` in this function, as it works solely on strings, I opted to keep it decoupled from the dependency. What do you think?

Reply:
it just caught my attention.. I think we should indeed use the proper tool for each task.. =)